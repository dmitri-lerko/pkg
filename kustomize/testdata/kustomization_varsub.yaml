---
apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
kind: Kustomization
metadata:
  name: app
  namespace: apps
spec:
  interval: 4m0s
  path: ./kustomize
  prune: true
  sourceRef:
    kind: GitRepository
    name: app
  validation: client
  postBuild:
    substitute:
      cluster_env: "prod"
      cluster_region: "eu-central-1"
    substituteFrom:
      - kind: ConfigMap
        name: app-vars
