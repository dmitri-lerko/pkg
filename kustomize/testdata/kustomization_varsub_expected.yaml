apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    environment: prod
    region: eu-central-1
  name: app
  namespace: apps
spec:
  selector:
    matchLabels:
      app: app
  template:
    metadata:
      annotations:
        prometheus.io/port: "9797"
        prometheus.io/scrape: "true"
      labels:
        app: app
    spec:
      containers:
      - image: nginx:1.21.4
        imagePullPolicy: IfNotPresent
        name: nginx
        resources:
          limits:
            cpu: 2000m
            memory: 512Mi
          requests:
            cpu: 100m
            memory: 64Mi
---
apiVersion: v1
data:
  image: ${image}
  prometheus_port: ${prometheus_port}
  prometheus_scrape: ${prometheus_scrape:=false}
kind: ConfigMap
metadata:
  labels:
    environment: ${cluster_env:=dev}
    kustomize.toolkit.fluxcd.io/substitute: disabled
    region: ${cluster_region}
  name: app-vars
  namespace: apps
